mislabeled/probe/_linear.py

```diff
@@ -82,37 +175,76 @@ def linearize_pipeline(estimator, X, y):
     return linearize(estimator[-1], X, y)
 
 
-@linearize.register(LogisticRegression)
-@linearize.register(LogisticRegressionCV)
-@linearize.register(SGDClassifier)
-@linearize.register(ElasticNet)
-@linearize.register(ElasticNetCV)
-@linearize.register(Lasso)
-@linearize.register(LassoCV)
-@linearize.register(RidgeClassifier)
-@linearize.register(RidgeClassifierCV)
-@linearize.register(LinearSVC)
+@linearize.register(SGDRegressor)
 @linearize.register(Ridge)
 @linearize.register(RidgeCV)
-@linearize.register(LinearRegressor)
-@linearize.register(LinearSVR)
-@linearize.register(SGDRegressor)
-def linearize_linear_model(estimator, X, y):
+@linearize.register(RidgeClassifier)
+def linearize_linear_model_ridge(estimator, X, y):
     X, y = check_X_y(X, y, accept_sparse=True, dtype=[np.float64, np.float32])
     coef = estimator.coef_.T
-    intercept = estimator.intercept_
+    intercept = estimator.intercept_ if estimator.fit_intercept else None
     if is_classifier(estimator):
-        if coef.ndim > 1 and coef.shape[1] == 1:
-            coef = np.hstack((-coef, coef))
-        linear = LinearClassifier(coef, intercept)
+        lb = LabelBinarizer(pos_label=1, neg_label=-1)
+        y = lb.fit_transform(y)
+
+    if hasattr(estimator, "alpha_"):
+        regul = estimator.alpha_
     else:
-        if coef.ndim > 1 and coef.shape[1] == 1:
-            coef = coef.ravel()
-            intercept = intercept.item()
-        linear = LinearRegressor(coef, intercept)
+        regul = estimator.alpha
+
+    linear = LinearModel(coef, intercept, loss="l2", regul=regul)
     return linear, X, y
 
 
+@linearize.register(SGDClassifier)
+def linearize_linear_model_sgdclassifier(estimator, X, y):
+    X, y = check_X_y(X, y, accept_sparse=False, dtype=[np.float64, np.float32])
+    coef = estimator.coef_.T
+    intercept = estimator.intercept_ if estimator.fit_intercept else None
+    linear = LinearModel(coef, intercept, loss=estimator.loss, regul=estimator.alpha)
+    return linear, X, y
+
+
+@linearize.register(LogisticRegression)
+@linearize.register(LogisticRegressionCV)
+def linearize_linear_model_logreg(estimator, X, y):
+    X, y = check_X_y(X, y, accept_sparse=True, dtype=[np.float64, np.float32])
+    coef = estimator.coef_.T
+    intercept = estimator.intercept_ if estimator.fit_intercept else None
+    if hasattr(estimator, "C_"):
+        regul = 1.0 / estimator.C_
+    else:
+        regul = 1.0 / estimator.C
+    linear = LinearModel(coef, intercept, loss="log_loss", regul=regul)
+    return linear, X, y
+
+
+# commented out while they are not used in tests
+# Lasso and Elasticnet are yet to be added
+# @linearize.register(ElasticNet)
+# @linearize.register(ElasticNetCV)
+# @linearize.register(Lasso)
+# @linearize.register(LassoCV)
+# @linearize.register(RidgeClassifierCV)
```

Comment:
Should RidgeClassifierCV be uncommented ?

Reply:
I commented out all classifiers that were not used in the tests